internal/venice-common/src/main/java/com/linkedin/venice/pushmonitor/PushStatusDecider.java

```diff
@@ -274,23 +274,25 @@ protected ExecutionStatus getPartitionStatus(
     // partitions can be completed. Vice versa, partitions will be in error state if leader is in error
     // state.
     boolean isLeaderCompleted = true;
-    int previouslyDisabledReplica = 0;
+    int previouslyDisabledErrorReplica = 0;
     for (Map.Entry<Instance, String> entry: instanceToStateMap.entrySet()) {
       ExecutionStatus currentStatus =
           getReplicaCurrentStatus(partitionStatus.getReplicaHistoricStatusList(entry.getKey().getNodeId()));
       if (entry.getValue().equals(HelixState.LEADER_STATE)) {
         if (!currentStatus.equals(COMPLETED)) {
           isLeaderCompleted = false;
         }
-
-        if (currentStatus.equals(ERROR) && callback != null) {
+        if (currentStatus.equals(ERROR) && callback != null
+            && !callback.isReplicaDisabled(entry.getKey().getNodeId(), partitionStatus.getPartitionId())) {
           callback.disableReplica(entry.getKey().getNodeId(), partitionStatus.getPartitionId());
         }
       } else if (entry.getValue().equals(HelixState.OFFLINE_STATE)) {
         // If the replica is in offline state, check if its due to previously disabled replica or not.
         if (callback != null
             && callback.isReplicaDisabled(entry.getKey().getNodeId(), partitionStatus.getPartitionId())) {
-          previouslyDisabledReplica++;
+          if (!currentStatus.equals(ERROR)) { // Dont count previous replica if it is not error status.
+            previouslyDisabledErrorReplica++;
```

Comment:
Should we skip the status counting if the helix state is `offline`? and I think logical-wise, it makes more sense.
The ultimate result is same, but it will improve the readability.

Reply:
as discussed offline, I have updated this logic